### PR TITLE
General improvements/fixes for PAYG indicator

### PR DESCRIPTION
**Pull request by andrunko**
This includes the following updates:
- Always keep status in sync when session mode changes (e.g. the menu sensitivity is correctly updated after unlocking the screen if the lock code was entered while the screen was locked)
- Always show account id menu item if supported
- Properly handle destroy (The parent class SystemIndicator doesn't define '_onDestroy')
- Reuse existing payg manager instance

https://phabricator.endlessm.com/T30647

### CODE REVIEW

File: js/ui/status/payg.js [hunk 1]
```diff
@@ -64,21 +70,35 @@ class PaygIndicator extends PanelMenu.SystemIndicator {
...
+        this._sessionModeUpdatedId = Main.sessionMode.connect('updated',
+            this._sync.bind(this));
+
+        this.connect('destroy', this._onDestroy.bind(this));
+
+        this._sync();
     }
 
     _onDestroy() {
-        super._onDestroy();
```
Comment (GeorgesStavracas) [inline js/ui/status/payg.js:75]: You still need to preserve this chainup, don't you?

Comment (andrunko) [reply js/ui/status/payg.js:75]: Not really, this was actually wrong and this method never invoked as the base class doesn't define `_onDestroy()` or connects to the destroy signal.

Comment (andrunko) [reply js/ui/status/payg.js:75]: The base class being `PanelMenu.SystemIndicator` which inherits from `St.BoxLayout`.

Comment (GeorgesStavracas) [reply js/ui/status/payg.js:75]: Okay

**Pull request merged**